table/block_based/block_based_table_reader.cc

```diff
@@ -386,9 +386,20 @@ Cache::Handle* BlockBasedTable::GetEntryFromCache(
     const Cache::CacheItemHelper* cache_helper,
     const Cache::CreateCallback& create_cb, Cache::Priority priority) const {
   Cache::Handle* cache_handle = nullptr;
+
   if (cache_tier == CacheTier::kNonVolatileBlockTier) {
+    SetPerfLevel(kEnableCount);
+    PerfContext* perf_ctx = get_perf_context();
+    perf_ctx->Reset();
```

Comment:
Instead of resetting, its better to cache the counter values and calculate the difference after the call. The PerfContext may also be looked at by the upper layer, so resetting will clear the counters they're interested in.